src/multisig/proposeTransferExecuteContractInput.ts

```diff
@@ -72,19 +67,22 @@ export class ProposeTransferExecuteContractInput {
             arguments: options.arguments,
             nativeTransferAmount: 0n,
         });
-        const functionCallParts = Buffer.from(transaction.data).toString().split(ARGUMENTS_SEPARATOR);
-        const functionName = functionCallParts[0];
-        const functionArguments = [];
-        for (let index = 1; index < functionCallParts.length; index++) {
-            const element = functionCallParts[index];
-            functionArguments.push(element.valueOf());
-        }
-        const functionCall = [new BytesValue(Buffer.from(utf8ToHex(functionName))), ...functionArguments];
+
+        const functionCall = ProposeTransferExecuteContractInput.getFunctionCall(transaction);
+
         return new ProposeTransferExecuteContractInput({
             multisigContract: options.multisig,
             to: options.to,
             functionCall: functionCall,
             gasLimit: options.optGasLimit,
         });
     }
+
+    private static getFunctionCall(transaction: Transaction) {
```

Comment:
We should add a comment to explain the workaround / hack.

> In the multisig transactions factory, we leverage the generic smart contract transactions factory to properly encode the arguments (given the ABI) for the "other" contract execute proposal. However, since we have to handle them as `variadic<bytes>` when preparing the "top-level" multisig transaction, here we split back the data.

To-do for a future PR: convert the arguments to a list of buffers using the `ArgSerializer` or a `SmartContractTransactionsFactory.argsToDataParts()`, instead of the _<create transaction using factory, then parse back into list of buffers>_ hack.

Reply:
As discussed, this is ok for the moment as long as we add the comment. Will perhaps refactor later if needed.